File: backend/app/schemas/exercise.py
from pydantic import BaseModel


class ExerciseResponse(BaseModel):
    id: str
    canonical_key: str
    name_en: str
    name_he: str
    description_en: str | None = None
    description_he: str | None = None
    difficulty_level: str
    body_focus: list[str]
    position: str | None = None
    possible_equipment: list[str]
    intensity: int | None = None
    min_duration_seconds: int | None = None
    max_duration_seconds: int | None = None
    min_reps: int | None = None
    max_reps: int | None = None
    phase_affinities: list[str]
    teaching_cues_en: list[str]
    teaching_cues_he: list[str]
    common_mistakes_en: list[str]
    common_mistakes_he: list[str]
    aliases_en: list[str]
    aliases_he: list[str]
    pilates_principles: list[str]
    is_active: bool


class ExerciseListResponse(BaseModel):
    exercises: list[ExerciseResponse]
    count: int

File: mobile/src/types/exercise.ts
export interface Exercise {
  id: string;
  canonical_key: string;
  name_en: string;
  name_he: string;
  description_en: string | null;
  description_he: string | null;
  difficulty_level: string;
  body_focus: string[];
  position: string | null;
  possible_equipment: string[];
  intensity: number | null;
  min_duration_seconds: number | null;
  max_duration_seconds: number | null;
  min_reps: number | null;
  max_reps: number | null;
  phase_affinities: string[];
  teaching_cues_en: string[];
  teaching_cues_he: string[];
  common_mistakes_en: string[];
  common_mistakes_he: string[];
  aliases_en: string[];
  aliases_he: string[];
  pilates_principles: string[];
  is_active: boolean;
}
